deltachat-jsonrpc/src/api/mod.rs

```diff
@@ -1336,6 +1337,40 @@ impl CommandApi {
         .await
     }
 
+    /// Starts to provide the backup for remote devices to retrieve.
+    ///
+    /// Can be cancelled by stopping the ongoing process.  Success or failure can be tracked
+    /// via the `ImexProgress` event which should either reach `1000` for success or `0` for
+    /// failure.
+    ///
+    /// This **stops IO**.  After completion `start_io` must be called to restart IO.
+    ///
+    /// Returns the QR code as a rendered SVG image.
+    async fn provide_backup(&self, account_id: u32, path: String) -> Result<String> {
+        let ctx = self.get_context(account_id).await?;
+        ctx.stop_io().await;
+        let provider = imex::BackupProvider::prepare(&ctx, Path::new(&path)).await?;
+        let qr = provider.qr();
+        let svg = match generate_backup_qr(&ctx, &qr).await {
+            Ok(svg) => svg,
+            Err(err) => {
+                ctx.stop_ongoing().await;
+                return Err(err);
+            }
+        };
+        Ok(svg)
+    }
+
+    /// Gets a backup from a remote provider.
+    ///
+    /// Can be cancelled by stopping the ongoing process.
+    async fn get_backup(&self, account_id: u32, qr_text: String) -> Result<()> {
```

Comment:
@Simon-Laux @link2xt This API blocks for the duration of the transfer, yet it can be cancelled via the ongoing process.  Is that fine behaviour for JSON-RPC?